test/unistd/links.c

```diff
@@ -31,10 +31,11 @@ void setup() {
     FS.mkdir('folder');
```

Comment:
I would like to see some tests or absolute symlinks added to `test/unistd/links.c`.

`test/unistd/links.c` it where we run as many of the symlink tests as possible.  `symlink_on_nodefs` I think should be reserved to tests that we can't run outside of nodefs (i.e. tests that involve using non-trivial mounting setups that require nodefs to pass).

One reason that we should use `links.c` where possible is that its actually possible to run `links.c` on native desktop systems to confirm they have the same behaviour as emscripten (i.e. to check emscripten is doing the right thing). 

Perhaps `symlink_on_nodefs.c` should be renamed to something better.. but I can't think of anything.    Basically its for tests that can't run outside of nodefs.

Reply:
@sbc100 Ok, I understand. I should probably duplicate the tests : 

```
test_relative_path_symlinks();
test_absolute_path_symlinks();
```
 from `symlink_on_nodefs.c` based on the symlink system from `links.c` in `setup()` function.
 
 Would you like me to refactor the code readability in `links.c` as well like you indicated below : 
 
 ```
 int main() {
   setup();
   test_foo_bar();
   test_baz();
   return 0;
}
```

Reply:
I think so yes, but maybe wait for #21890 to land to avoid too many conflicts.

Reply:
Actually I wonder if we even need `symlink_on_nodefs.c` at all anymore?  Since it won't really be testing anything will it?   Other than "symlinks to outside the VFS don't work" which is basically the same as "symlinks to non-existant files don't work"?  Or is it will worth keeping it around do you think?

Reply:
My understanding is that this use case and this test could probably be an indicator that this is not possible and that it shouldn't be possible. So we could imagine that the only test that the `symlink_on_nodefs.c` could have are : 

```
  test_inside_symlink();
  test_outside_symlink();
```

The others will be moved to `links.c` ? 

What I find mostly important here is the proof that this has been decided.